websites/C/CodinGame/presence.ts

```diff
@@ -0,0 +1,199 @@
+const presence = new Presence({
+		clientId: "991078618866274364",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/ar0zfXY.png",
+			startTimestamp: browsingTimestamp,
+		},
+		path = document.location.pathname;
+	if (path === "/") presenceData.details = "Browsing...";
+	else if (path.includes("/home")) {
+		presenceData.details = "On dashboard";
+		presenceData.state = `${
+			document.querySelectorAll(
+				"div[class='home-content'] > div > div > div > div > home > div > home-react > div > div > section > section > div > div > div > div"
+			)[0].textContent
+		} - Rank : ${
+			document.querySelectorAll(
+				"div[class='home-content'] > div > div > div > div > home > div > home-react > div > div > section > section > div > div > div > div"
+			)[4].textContent
+		}`;
+	} else if (path.includes("/training")) {
+		if (path.includes("/training/easy/")) {
+			presenceData.details = "Training: Easy";
+			presenceData.state = `${
+				document.querySelector("h1[class='title']").textContent
+			} (${document.querySelector("div[class='chart-inner']").textContent}%)`;
+		} else if (path.includes("/training/medium/")) {
+			presenceData.details = "Training: Medium";
+			presenceData.state = `${
+				document.querySelector("h1[class='title']").textContent
+			} (${document.querySelector("div[class='chart-inner']").textContent}%)`;
+		} else if (path.includes("/training/hard/")) {
+			presenceData.details = "Training: Hard";
+			presenceData.state = `${
+				document.querySelector("h1[class='title']").textContent
+			} (${document.querySelector("div[class='chart-inner']").textContent}%)`;
+		} else if (path.includes("/training/expert/")) {
+			presenceData.details = "Training: Very Hard";
+			presenceData.state = `${
+				document.querySelector("h1[class='title']").textContent
+			} (${document.querySelector("div[class='chart-inner']").textContent}%)`;
```

Comment:
This is repeated with the only difference of the training's difficulty, instead check if the path includes `/training/` and if it does, take the difficulty from the URL by using `document.location.pathname.split("/")[n]`, where `n` is the index value